vars/publishToNpm.groovy

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to publish artifacts to NPM registry under @opensearch-project namespace
+@param Map args = [:] args A map of the following parameters
+@param args.repository <required> - Repository to be used to publish the artifact to npm
+@param args.tag <required> - Tag reference to be used to publish the artifact
+*/
+void call(Map args = [:]) {
+
+    checkout([$class: 'GitSCM', branches: [[name: "${args.tag}" ]], userRemoteConfigs: [[url: "${args.repository}" ]]])
+    
+    withCredentials([string(credentialsId: 'publish-to-npm-token', variable: 'NPM_TOKEN')]){
+        sh """npm set registry "https://registry.npmjs.org"; npm set //registry.npmjs.org/:_authToken ${NPM_TOKEN}; npm publish --dry-run && npm publish --access public"""
```

Comment:
Maybe we could add one another command `npm logout` at the beginning to make sure we don't accidentally publish to somewhere else or this could be replaced with new login?

Reply:
We are using a token and not logging into the account so no need to log out. The scope of scope is within the `withCredentials`.